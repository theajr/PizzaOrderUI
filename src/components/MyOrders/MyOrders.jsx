import React, { useEffect, useState } from 'react';
import axios from '../../util/axios';
import { LOAD_MY_ORDERS } from '../../util/constants';
import OrderSummary from '../OrderSummary/OrderSummary';
import { connect } from '../../util/appContext';
import Loader from '../Loader';

const MyOrders = props => {
  const [fetching, setFetching] = useState(!false);
  const { updateMyOrders, orders } = props;
  useEffect(() => {
    setFetching(true);
    axios
      .get('/orders', { params: {} })
      .then(res => {
        console.log('----', res);
        updateMyOrders(res.data);
      })
      .catch(e => console.warn)
      .finally(() => setFetching(false));
    return () => {};
  }, []);
  return (
    <Loader message="Fetching your orders.." show={fetching}>
      <div>
        {!orders.length && <h3>You didn't place any orders, yet!</h3>}
        {!fetching &&
          orders.map(order => <OrderSummary key={order.id} info={order} />)}
      </div>
    </Loader>
  );
};

const mapStateToProps = state => {
  const { orders = [] } = state?.auth?.profile || {};

  return {
    orders,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateMyOrders: orders =>
      dispatch({ type: LOAD_MY_ORDERS, payload: orders }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MyOrders);
