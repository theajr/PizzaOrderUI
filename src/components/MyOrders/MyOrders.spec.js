import React from 'react';
import { shallow } from 'enzyme';
import MyOrders from './MyOrders';
import { renderComp } from '../../../tests/utilRender';

describe('<MyOrders />', () => {
  test('renders', () => {
    const wrapper = renderComp(<MyOrders />);
    expect(wrapper).toMatchSnapshot();
  });
});
