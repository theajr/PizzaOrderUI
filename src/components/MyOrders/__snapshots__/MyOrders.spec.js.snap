// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyOrders /> renders 1`] = `ShallowWrapper {}`;
